### ISSUE TITLE
Che Theia Web View doesn't work properly when a plugin is running in its own sidecar

### ISSUE THREAD

**Issue by azatsarynnyy** [eclipse-che/che#16870]
### Describe the bug
Che Theia Web View doesn't work properly when a plugin is running in its own sidecar.

### Steps to reproduce

1. Start a Workspace with the following plugin descriptor:
```yaml
apiVersion: v2
publisher: redhat
name: vscode-didact
version: 0.1.14
type: VS Code extension
displayName: YAML
title: Didact Tutorial Tools by Red Hat
description: Provides interactive tutorial support to Visual Studio Code, via didact links calling VS Code commands, Markdown, AsciiDoc, and the VS Code webview.
icon: https://www.eclipse.org/che/images/logo-eclipseche.svg
repository: https://github.com/redhat-developer/vscode-didact
category: Other
firstPublicationDate: "2020-05-06"
spec:
  containers:
    - image: "quay.io/eclipse/che-sidecar-node:10-0cb5d78"
      name: vscode-didact
      memoryLimit: "256Mi"
  extensions:
    - https://github.com/redhat-developer/vscode-didact/releases/download/0.1.14/vscode-didact-0.1.14.vsix
```

2. Go to `DIDACT TUTORIALS` view and run `Didact Demo`
![image](https://user-images.githubusercontent.com/1636395/81304200-6c491380-9085-11ea-9591-b7e5495f817f.png)

3. The page is opened in Theia Web View has no styles applied:
![image](https://user-images.githubusercontent.com/1636395/81307427-51789e00-9089-11ea-98e7-bdeab64f0e77.png)

Also, the links that calling the commands aren't working.

### Expected behavior
The page is opened in Theia Web View should have the styles applied:
![image](https://user-images.githubusercontent.com/1636395/81307951-e4b1d380-9089-11ea-86ef-f12cc56446f7.png)

The links that calling the commands should work.

### Additional context
From the browser's dev console:
```
main.js:469 GET https://routegp3bizjr-azatsary-che.8a09.starter-us-east-2.openshiftapps.com/webview/theia-resource/file/tmp/vscode-unpacked/redhat.vscode-didact.0.1.14.mnjqnseceg.vscode-didact-0.1.14.vsix/extension/media/webviewslim.css net::ERR_ABORTED 404
(anonymous) @ main.js:469
setTimeout (async)
(anonymous) @ main.js:466
main.js:469 GET https://routegp3bizjr-azatsary-che.8a09.starter-us-east-2.openshiftapps.com/webview/theia-resource/file/tmp/vscode-unpacked/redhat.vscode-didact.0.1.14.mnjqnseceg.vscode-didact-0.1.14.vsix/extension/media/main.js net::ERR_ABORTED 404
```

### Sub-tasks
- [x] Implement simple file system provider to fetch files from containerized plugin https://github.com/eclipse/che/issues/18405

**Comment by azatsarynnyy** [comment]
I think retrieving the plugin resources for WebView can be done the similar way as we've fixed fetching the icons for the contributed Views https://github.com/eclipse/che/issues/13750 in https://github.com/eclipse/che-theia/pull/430

**Comment by ericwill** [comment]
It would be great if this could make it into 7.22, as there are a number of extensions that are facing this issue.

**Comment by gattytto** [comment]
sometimes the files are downloaded/provided by the extension at runtime, but in other scenarios files are already there from the Dockerfile build process..

**Comment by gattytto** [comment]
EFanZh/vsextensions/graphviz-preview graphViz Preview needed by DAML extension (WIP)
joaompinto/vsextensions/vscode-graphviz yet another graphviz viewer

are 2 more extensions blocked by this issue. 
![image](https://user-images.githubusercontent.com/407397/96965068-ebba8a00-14e1-11eb-914d-a2ea4c92983e.png)

**Comment by azatsarynnyy** [comment]
Thanks @gattytto for providing a use case. We plan to start investing it next month.

**Comment by gattytto** [comment]
"special" usecase is haskell which comes with DOCS from the sidecar setup:
![image](https://user-images.githubusercontent.com/407397/97035711-1dad0a00-153d-11eb-8705-b52157e204ed.png)

**Comment by azatsarynnyy** [comment]
@vzhukovs FYI: there's upstream Theia PR https://github.com/eclipse-theia/theia/pull/8468 that seems addresses a similar issue with fetching Plug-in resources but related to CodeSnippets files.

**Comment by azatsarynnyy** [comment]
It's fixed in the following PRs: https://github.com/eclipse/che-theia/pull/947, https://github.com/eclipse/che-theia/pull/946, https://github.com/eclipse-theia/theia/pull/8832, https://github.com/eclipse-theia/theia/pull/8833

**Comment by nsharma1989** [comment]
Hi, I am facing another issue which is related to the above problem. My plugin runs in a sidecar and uses a web view. The issue is, the web view only works once in an active session and if you close the webview and open again it doesn't load the resources. The below is the URL links to the resources from dev tools
```
[When view is opened for first time]
 https://CHE_THEIA_URL/webview/theia-resource/file-sidecar-EXT_ID/tmp/vscode-unpacked/EXTENSION.vsix/extension/dist/styles.cs
 
[When it never opens]
 https://CHE_THEIA_URL/webview/theia-resource/file///tmp/vscode-unpacked/iEXTENSION.vsix.vsix/extension/dist/styles.css
```
